models/build_request.go

```diff
@@ -1,22 +1,98 @@
 package models
 
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+)
+
 // BuildRequest struct
+// TODO: Convert the return sms and wait return sms to boolean
 type BuildRequest struct {
-	SMSID         string `json:"smsid"`
-	SMSTitle      string `json:"smstitle"`
-	SMSText       string `json:"smstext"`
-	MobileNumber  string `json:"mobilenumber"`
-	OriginalURL   string `json:"originalurl"`
-	OriginalURL2  string `json:"originalurl2"`
-	OriginalURL3  string `json:"originalurl3"`
-	OriginalURL4  string `json:"originalurl4"`
-	Status        string `json:"status"`
-	ReturnSMS     string `json:"returnsms"`
-	WaitReturnSMS string `json:"waitreturnsms"`
-	Type          string `json:"type"`
-	Au            string `json:"au"`
-	Docomo        string `json:"docomo"`
-	Softbank      string `json:"softbank"`
-	Gateway       string `json:"gateway"`
-	Rakuten       string `json:"rakuten"`
+	SMSID         string
+	SMSTitle      string
+	SMSText       string
+	MobileNumber  string
+	OriginalURL   string
+	OriginalURL2  string
+	OriginalURL3  string
+	OriginalURL4  string
+	Status        string
+	ReturnSMS     string
+	WaitReturnSMS string
+	Type          string
+	Au            AuConfig
+	Docomo        DocomoConfig
+	Softbank      SoftbankConfig
+	Gateway       GatewayConfig
+	Rakuten       RakutenConfig
+	Sim           SimConfig
+}
+
+// NewBuildRequest returns ...
+func NewBuildRequest() BuildRequest {
+	return BuildRequest{
+		Au:       AuConfig{NumberOfRetries: 1},
+		Docomo:   DocomoConfig{NumberOfRetries: 1, RetryInterval: []int{0}},
+		Softbank: SoftbankConfig{NumberOfRetries: 1, RetryInterval: []int{5}},
+		Gateway:  GatewayConfig{NumberOfRetries: 1, RetryInterval: []int{1}},
+		Rakuten:  RakutenConfig{NumberOfRetries: 1, RetryInterval: []int{2}},
+		Sim:      SimConfig{NumberOfRetries: 1},
+	}
+}
+
+// MarshalJSON overrides the custom serialization to prepare the data for the MediaSms API
+func (br BuildRequest) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&struct {
+		SMSID         string `json:"smsid,omitempty"`
+		SMSTitle      string `json:"smstitle,omitempty"`
+		SMSText       string `json:"smstext"`
+		OriginalURL   string `json:"originalurl,omitempty"`
+		OriginalURL2  string `json:"originalurl2,omitempty"`
+		OriginalURL3  string `json:"originalurl3,omitempty"`
+		OriginalURL4  string `json:"originalurl4,omitempty"`
+		MobileNumber  string `json:"mobilenumber"`
+		Status        string `json:"status,omitempty"`
+		ReturnSMS     string `json:"returnsms,omitempty"`
+		WaitReturnSMS string `json:"waitreturnsms,omitempty"`
+		Type          string `json:"type,omitempty"`
+		Au            string `json:"au,omitempty"`
+		Docomo        string `json:"docomo,omitempty"`
+		Softbank      string `json:"softbank,omitempty"`
+		Gateway       string `json:"gateway,omitempty"`
+		Rakuten       string `json:"rakuten,omitempty"`
+		Sim           string `json:"sim,omitempty"`
```

Comment:
Where sim is written on the documentation ? I couldn't find this request parameter on media4u doc.

Reply:
Its on the english documentation page 15 
![Uploading Screen Shot 2020-11-13 at 11.28.01.png…]()

Reply:
Thank you for sharing. It seems not to be on Japanese documentation. I'm gonna ask media4u what is the meaning of this option and we need to use it.

Reply:
About sim, Media4u confirmed English documentation was correct. They added sim parameter to Japanese documentation but I'm still asking about what is the meaning of sim parameter.